pkg/client/restclient/config.go

```diff
@@ -130,10 +130,9 @@ type ContentConfig struct {
 	// a RESTClient directly. When initializing a Client, will be set with the default
 	// code version.
 	GroupVersion *unversioned.GroupVersion
-	// Codec specifies the encoding and decoding behavior for runtime.Objects passed
-	// to a RESTClient or Client. Required when initializing a RESTClient, optional
-	// when initializing a Client.
-	Codec runtime.Codec
+	// NegotiatedSerializer is used for obtaining encoders and decoders for multiple
+	// supported media types.
+	NegotiatedSerializer runtime.NegotiatedSerializer
```

Comment:
So to clarify - we have now different serializer to standard operations, different fo watch, different for pretty-printing, probably more in the future. So NegotiatedSerializer in my opinion is the easiest way to avoid passing a bunch of different arguments evewhere.

Regarding leaving Codec - how would that be simpler? We are setting all of this automatically, so I think there is no point in leaving it. [Also this is a step towards removing LegacySerializer, which I think we want to do at some point].